realm/src/main/java/io/realm/internal/TableView.java

```diff
@@ -867,18 +866,18 @@ public void sort(long columnIndex) {
         nativeSort(nativePtr, columnIndex, true);
     }
 
-    protected native void nativeSort(long nativeTableViewPtr, long columnIndex, boolean ascending);
+    protected native void nativeSort(long nativeTableViewPtr, long columnIndex, boolean sortOrder);
 
-    public void sort(List<Long> columnIndices, List<Order> order) {
+    public void sort(List<Long> columnIndices, Sort[] sortOrders) {
```

Comment:
It has to be a list as each `SortOrder` correspond to the same entry in `columnIndices` (for multi-field sorting), so a Set wouldn't really work in this case.